### ISSUE TITLE
[8.4.0+] Wrapper for React doesn't work on CodeSandbox with `react@17`

### ISSUE THREAD

**Issue by mrpiotr-dev** [handsontable/handsontable#8842]
### Description
The problem occurs only on codesandbox, which is a base platform to run examples from our documentation and pull requests. Fortunately,  locally created `create-react-app` and stackblitz-hosted examples work correctly.

### Steps to reproduce
1. Go to any codesandbox-hosted example from the [`Using the HotColumn component`](https://handsontable.com/docs/react-hot-column) guide
2. Update `react` and `react-dom` to v17.x

Result:
```
Error
Objects are not valid as a React child (found: object with keys {$$typeof, type, key, ref, props, _owner, _store}). If you meant to render a collection of children, use an array instead.
```
<details>
  <summary>Screenshot</summary>

  ![image](https://user-images.githubusercontent.com/11849621/136186861-b1957b38-782c-47d3-91d5-f544d77ad114.png)
</details>

### Demo
https://codesandbox.io/s/declaring-column-settings-forked-8465e?file=/src/index.js

### Your environment
* Handsontable version: 8.4.0+
* Browser Name and version: any
* Operating System: any

**Comment by akoskm** [comment]
Thanks for writing up this issue, I'm experiencing the same problem.
Adding @handsontable/react and importing
```
import { HotTable } from "@handsontable/react";
```
breaks the sandbox.

**Comment by jansiegel** [comment]
Looks like it's caused by a bug in `core-js`, which breaks a feature-detecting logic on CodeSandbox's side.

From what I figured, our demo breaks, because it uses a `core-js` polyfill for `Symbol` (which results in React not recognizing component elements properly, as their `$$typeof` property is being set to something unexpected - a `Symbol`-like object).

_That_ happens because of a [bug introduced in `core-js@3.12`](https://github.com/codesandbox/codesandbox-client/issues/5943), which fails to recognize the `v8` version in some environments (including CodeSandbox's). This problem [was fixed in `core-js@3.19`](https://github.com/zloirock/core-js/blob/master/CHANGELOG.md#3190---20211025).

Seems to me like the example works in `8.3.2`, bot not `8.4.0` because of some `npm` logic, which resolves our `^3.0.0` entry as `3.9.1` for `8.3.2`, but `3.12.1+` for `8.4.0+`.
The upcoming `handsontable@11.0` has a `core-js@3.19.1` entry in its `package-lock`, so I wouldn't be surprised if this problem would fix itself magically just by releasing `11.0.0`.

**So, tl;dr;** 
Adding `core-js@3.19+` as a dependency in CodeSandbox clears this error: https://codesandbox.io/s/declaring-column-settings-forked-zq8e1?file=/package.json:295-320

The alternative resolution would be bumping the minimal `core-js` version in the dependencies of `handsontable`, which should most probably end up with the same results.

WDYT @wojciechczerniak @warpech?

**Comment by wojciechczerniak** [comment]
Do we hava a build from Codesandbox CI (any latest PR) that we can test this scenario with and confirm it's really solved for develop/v11?

**Comment by jansiegel** [comment]
> Do we hava a build from Codesandbox CI (any latest PR) that we can test this scenario with and confirm it's really solved for develop/v11?

Great idea - we don't have a Codesandbox CI build of the exact state of the release branch (I can make a PR draft, if needed), but I checked it with [a `develop`'s build from a few days ago](https://github.com/handsontable/handsontable/pull/8907), and it seems to work fine: https://codesandbox.io/s/declaring-column-settings-forked-50959.

**Comment by wojciechczerniak** [comment]
> > Do we hava a build from Codesandbox CI (any latest PR) that we can test this scenario with and confirm it's really solved for develop/v11?
> 
> Great idea - we don't have a Codesandbox CI build of the exact state of the release branch (I can make a PR draft, if needed), but I checked it with [a `develop`'s build from a few days ago](https://github.com/handsontable/handsontable/pull/8907), and it seems to work fine: https://codesandbox.io/s/declaring-column-settings-forked-50959.

We may confirm with PR if needed, but with your links I'm now 99% sure it's ok.

**Comment by warpech** [comment]
Thank you @jansiegel. Then we can remove the issue from the scope of 11.0.0 based on your research in https://codesandbox.io/s/declaring-column-settings-forked-50959, if you are confident that is fixed there.

Later, the broken examples for version 8.x/9.x/10.x can be solved with smaller urgency by adding `core-js@3.19+` as a dependency. It would be nice to add a comment at the top of the JS file that an additional `core-js@3.19+` dependency is used only to resolve a CodeSandbox issue with Handsontable 8.4.0-10.0.0 (fixed in 11.0.0).

**Comment by mrpiotr-dev** [comment]
Maybe that's a good starting point to discuss if `core-js` should be a part of the final bundle
- or -
there should be an additional `core-js`-free bundle.

**Comment by warpech** [comment]
Done in https://github.com/handsontable/handsontable/pull/8966, https://github.com/handsontable/handsontable/pull/8962

**Comment by AMBudnik** [comment]
Thank you for the reference @warpech. I'm closing this thread.
